mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearAlgorithm.scala

```diff
@@ -84,7 +87,20 @@ abstract class StreamingLinearAlgorithm[
     }
     data.foreachRDD { (rdd, time) =>
       if (!rdd.isEmpty) {
-        model = Some(algorithm.run(rdd, model.get.weights))
+        val newModel = algorithm.run(rdd, model.get.weights)
+
+        val numNewDataPoints = rdd.count()
+        val discount = getDiscount(numNewDataPoints)
+
+        val updatedDataWeight = previousDataWeight * discount + numNewDataPoints
+        val lambda = numNewDataPoints / math.max(updatedDataWeight, 1e-16)
```

Comment:
I actually think it's more clear if `StreamingDecay` had a `getLambda`, otherwise you will need to document how `discount` is calculated in `StreamingDecay` as well as how `lambda` affects the weight updates in `StreamingLinearAlgorithm`